contracts/managers/triggers/MovingAverageToAssetPriceCrossoverTrigger.sol

```diff
@@ -0,0 +1,239 @@
+/*
+    Copyright 2019 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity 0.5.7;
+pragma experimental "ABIEncoderV2";
+
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import { ITrigger } from "./ITrigger.sol";
+import { IOracle } from "../../meta-oracles/interfaces/IOracle.sol";
+import { IMetaOracleV2 } from "../../meta-oracles/interfaces/IMetaOracleV2.sol";
+
+
+/**
+ * @title MovingAverageToAssetPriceCrossoverTrigger
+ * @author Set Protocol
+ *
+ * Implementing the ITrigger interface, this contract is queried by a
+ * RebalancingSetToken Manager to determine if the market is in a bullish 
+ * state by checking if the the trading pair price is above or below a simple
+ * or exponential moving average.
+ *
+ * Below the moving average means the RebalancingSetToken should be in the
+ * quote asset, above the moving average the RebalancingSetToken should be
+ * in the base asset.
+ */
+contract MovingAverageToAssetPriceCrossoverTrigger is
+    ITrigger
+{
+    using SafeMath for uint256;
+
+    /* ============ State Variables ============ */
+    IMetaOracleV2 public movingAveragePriceFeedInstance;
+    IOracle public assetPairOracleInstance;
+    uint256 public movingAverageDays;
+    uint256 public triggerFlippedIndex;
+    bool private lastConfirmedState;
+
+    // Time to start of confirmation period in seconds
+    uint256 public signalConfirmationMinTime;
+    // Time to end of confirmation period in seconds
+    uint256 public signalConfirmationMaxTime;
+    uint256 public lastInitialTriggerTimestamp;
+
+    bool public requiresConfirmation = true;
```

Comment:
Will all trigger contracts have `requiresConfirmation`?

Reply:
Yes and it's more for us to be able to more efficiently update them from off-chain